scan.go

```diff
@@ -289,14 +320,26 @@ func (s *Scan) nextBatch() int {
 	if startKey == nil {
 		startKey = s.StartRow
 	}
-	rs := s.getData(startKey, 0)
+
+	// Notice: ignore error here.
+	// TODO: add error check, now only add a log.
+	rs, err := s.getData(startKey, 0)
+	if err != nil {
+		log.Errorf("scan next batch failed - [startKey=%q], %v", startKey, errors.ErrorStack(err))
+	}
+
 	// current region get 0 data, switch next region
-	if rs != nil && len(rs) == 0 && s.nextStartKey != nil {
-		rs = s.getData(s.nextStartKey, 0)
+	if len(rs) == 0 && s.nextStartKey != nil {
+		// TODO: add error check, now only add a log.
+		rs, err = s.getData(s.nextStartKey, 0)
+		if err != nil {
+			log.Errorf("scan next batch failed - [startKey=%q], %v", s.nextStartKey, errors.ErrorStack(err))
+		}
 	}
-	if rs == nil {
+	if len(rs) == 0 {
```

Comment:
I am a little confused, for `rs == nil` or `len(rs) == 0` are both having no result, but why the old code uses `rs == nil` for chek?